src/decoder/mod.rs

```diff
@@ -306,6 +307,8 @@ pub struct Reader<R: BufRead + Seek> {
     scratch_buffer: Vec<u8>,
     /// Whether `ImageEnd` was already reached by `fn finish`.
     finished: bool,
+    /// How to expand interlaced data we encounter.
+    interlace: Adam7Variant,
```

Comment:
This should be settable, and there should be a method of selecting `Adam7Variant::Splat`. The simple choice would be `DecodeOptions` but if this would be useful to change each frame, discuss.

Reply:
`DecodeOptions` seems like a reasonable choice.  I don't see a scenario where changing this setting per-frame would be useful.